### PR TITLE
feat: change function statement to class

### PR DESCRIPTION
**Pull request by xtx1130**
ref: https://github.com/fastify/fastify-error/pull/61#issuecomment-1046734623
It will make breaking change in this PR.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] ~~documentation is changed or added~~
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -93,14 +93,3 @@ test('FastifyError.toString returns code', t => {
   const err = new NewError()
   t.is(err.toString(), 'FastifyError [CODE]: foo')
 })
-
-test('Create the error without the new keyword', t => {
```
Comment (jsumners) [inline test.js:97]: Why is this test removed?

**Review by jsumners — CHANGES_REQUESTED** [review]

File: test.js (hunk 1 above)
Comment (climba03003) [reply test.js:97]: It moved to `class` implementation. It can't be able to use `Error()` for initialize.

Comment (xtx1130) [reply test.js:97]: We have discussed in https://github.com/fastify/fastify-error/pull/61#issuecomment-1046734623. Well, I think it needs to be discussed within the fastify organization and make a decision.

File: index.js [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
-    } else if (a) {
-      this.message = format(message, a)
-    } else {
-      this.message = message
+  class FastifyError extends Base {
+    constructor (a, b, c) {
+      super()
+      if (new.target === undefined) {
+        return new FastifyError(a, b, c)
+      }
```
[the remark this answers is not in the document]
Comment (xtx1130) [reply index.js]: PTAL.

**Comment by mcollina** [comment]
I'm sorry but we are not in agreement on this change.

**Comment by xtx1130** [comment]
> I'm sorry but we are not in agreement on this change.

OK, got it😁

**Pull request closed without merging**